gcloud/bigquery/query.py

```diff
@@ -0,0 +1,336 @@
+# Copyright 2015 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Define API Queries."""
+
+import six
+
+from gcloud.bigquery._helpers import _TypedProperty
+from gcloud.bigquery._helpers import _rows_from_json
+from gcloud.bigquery.dataset import Dataset
+from gcloud.bigquery.job import QueryJob
+from gcloud.bigquery.table import _parse_schema_resource
+
+
+class _SyncQueryConfiguration(object):
+    """User-settable configuration options for synchronous query jobs.
+
+    Values which are ``None`` -> server defaults.
+    """
+    _default_dataset = None
+    _max_results = None
+    _timeout_ms = None
+    _preserve_nulls = None
+    _use_query_cache = None
+
+
+class QueryResults(object):
```

Comment:
Per #1278, the class is renamed, so that it no longer describes itself as a "job": it also gains a `job` accessor, which returns a `QueryJob` instance that can be used to gain access to the full configuration / statistics / status for the underlying job.